apps/sms/test/unit/thread_ui_test.js

```diff
@@ -1958,75 +1964,76 @@ suite('thread_ui.js >', function() {
         id: 1,
         threadId: 1,
         timestamp: +someDateInThePast
-      });
+      }).then(() => {
+        var message = ThreadUI.container.querySelector('[data-message-id="1"]');
 
-      var message = ThreadUI.container.querySelector('[data-message-id="1"]');
-
-      assert.isNotNull(originalMessage);
-      assert.isNotNull(message);
-      assert.notEqual(message, originalMessage);
+        assert.isNotNull(originalMessage);
+        assert.isNotNull(message);
+        assert.notEqual(message, originalMessage);
+      }).then(done,done);
     });
 
-    test('inserts message at the beginning', function() {
+    test('inserts message at the beginning', function(done) {
       someDateInThePast.setHours(someDateInThePast.getHours() - 1);
 
       ThreadUI.appendMessage({
         id: 2,
         threadId: 1,
         timestamp: +someDateInThePast
-      });
+      }).then(() => {
+        var messageItemNodes = ThreadUI.container.querySelectorAll('.message');
 
-      var messageItemNodes = ThreadUI.container.querySelectorAll('.message');
-
-      assert.equal(messageItemNodes.length, 2);
-      assert.equal(+messageItemNodes[0].dataset.messageId, 2);
-      assert.equal(+messageItemNodes[1].dataset.messageId, 1);
+        assert.equal(messageItemNodes.length, 2);
+        assert.equal(+messageItemNodes[0].dataset.messageId, 2);
+        assert.equal(+messageItemNodes[1].dataset.messageId, 1);
+      }).then(done,done);
     });
 
-    test('inserts message at the right spot in the middle', function() {
+    test('inserts message at the right spot in the middle', function(done) {
       someDateInThePast.setHours(someDateInThePast.getHours() + 2);
       ThreadUI.appendMessage({
         id: 3,
         threadId: 1,
         timestamp: +someDateInThePast
-      });
-
-      someDateInThePast.setHours(someDateInThePast.getHours() - 1);
-      ThreadUI.appendMessage({
-        id: 2,
-        threadId: 1,
-        timestamp: +someDateInThePast
-      });
-
-      var messageItemNodes = ThreadUI.container.querySelectorAll('.message');
+      }).then(() => {
+        someDateInThePast.setHours(someDateInThePast.getHours() - 1);
+        ThreadUI.appendMessage({
```

Comment:
Sorry missed this during previous review, we need ```return``` here as well, otherwise test will pass even in case we have failures inside second ```ThreadUI.appendMessage``` callback